sound/soc/sof/intel/hda.c

```diff
@@ -1057,8 +1116,30 @@ static int hda_sdw_machine_select(struct snd_sof_dev *sdev)
 	 */
 	if (link_mask && !pdata->machine) {
 		mach = pdata->desc->alt_machines;
-		while (mach && mach->link_mask && mach->link_mask != link_mask)
-			mach++;
+		while (mach && mach->link_mask) {
```

Comment:
@bardliao this parts nags me a bit. Looking into the definition of snd_soc_acpi_intel_icl_sdw_machines[] in your previous patch, the array ends with a {} which means mach will likely not be null after you're gone  through the 3 items in the array but will accessing mach->link_mask run into an oops? I maybe wrong but have you tried a case where you fail to find a match after going through the mach list?

Reply:
@ranj063 You are right. The mach is not null. But I think {} is also an valid item with all elements are zreo(empty) of snd_soc_acpi_intel_cml_machines[], right? I tried a case where I fail to find a match after going through the mach list and it works as what we expect.

Reply:
this is fine, we've used this for a while now. When we hit the last {}, then the link_mask is zero.